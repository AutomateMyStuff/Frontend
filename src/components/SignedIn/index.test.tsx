import { render, screen, waitFor } from '@testing-library/react'
import { SignedIn } from './index'

jest.mock("next-auth/react", () => {
    const originalModule = jest.requireActual('next-auth/react');
    const mockSession = {
        expires: new Date(Date.now() + 2 * 86400).toISOString(),
        user: { username: "admin" }
    };
    return {
        __esModule: true,
        ...originalModule,
        useSession: jest.fn(() => {
            return { data: mockSession, status: 'authenticated' }  // return type is [] in v3 but changed to {} in v4
        }),
    };
});

describe('Components/SignedIn', () => {
    it('renders SignedIn and match snapshot', () => {
        const { container } = render(<SignedIn />);
        expect(container).toMatchSnapshot();
    })

    it('Should see the header in the screen', () => {
        render(<SignedIn />);
        expect(screen.getByRole('heading', { name: "Signed In" })).toBeInTheDocument();
    });

    it('Should see the Logout button', () => {
        render(<SignedIn />);
        expect(screen.getByRole('button', { name: "Sign out" })).toBeInTheDocument();
    });
});