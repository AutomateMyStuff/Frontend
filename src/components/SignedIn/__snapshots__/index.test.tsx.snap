// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/SignedIn renders SignIn 1`] = `
<div>
  <div
    class="signedin_container"
  >
    <div
      class="signedin_content"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0zm-3.97-3.03a.75.75 0 0 0-1.08.022L7.477 9.417 5.384 7.323a.75.75 0 0 0-1.06 1.06L6.97 11.03a.75.75 0 0 0 1.079-.02l3.992-4.99a.75.75 0 0 0-.01-1.05z"
        />
      </svg>
      <h1>
        Signed In
      </h1>
      <p>
        You're signed in
        <br />
        as 
      </p>
    </div>
    <button
      class="btn_signout"
    >
      Sign out
    </button>
    <div
      class="lottie_container"
    >
      <div>
        <svg
          height="640"
          preserveAspectRatio="xMidYMid meet"
          style="width: 100%; height: 100%; transform: translate3d(0,0,0);"
          viewBox="0 0 360 640"
          width="360"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <defs>
            <clippath
              id="__lottie_element_2"
            >
              <rect
                height="640"
                width="360"
                x="0"
                y="0"
              />
            </clippath>
          </defs>
          <g
            clip-path="url(#__lottie_element_2)"
          >
            <g
              style="display: none;"
            >
              <g>
                <path />
              </g>
            </g>
            <g
              style="display: none;"
            >
              <g>
                <path />
              </g>
            </g>
            <g
              style="display: none;"
            >
              <g>
                <path />
              </g>
            </g>
            <g
              style="display: none;"
            >
              <g>
                <path />
              </g>
            </g>
            <g
              style="display: none;"
            >
              <g>
                <path />
              </g>
            </g>
            <g
              style="display: none;"
            >
              <g>
                <path />
              </g>
            </g>
            <g
              style="display: none;"
            >
              <g>
                <path />
              </g>
            </g>
            <g
              style="display: none;"
            >
              <g>
                <path />
              </g>
            </g>
            <g
              style="display: none;"
            >
              <g>
                <path />
              </g>
            </g>
            <g
              style="display: none;"
            >
              <g>
                <path />
              </g>
            </g>
            <g
              style="display: none;"
            >
              <g>
                <path />
              </g>
            </g>
            <g
              style="display: none;"
            >
              <g>
                <path />
              </g>
            </g>
            <g
              style="display: none;"
            >
              <g>
                <path />
              </g>
            </g>
            <g
              style="display: none;"
            >
              <g>
                <path />
              </g>
            </g>
            <g
              style="display: none;"
            >
              <g>
                <path />
              </g>
            </g>
            <g
              style="display: none;"
            >
              <g>
                <path />
              </g>
            </g>
            <g
              style="display: none;"
            >
              <g>
                <path />
              </g>
            </g>
            <g
              style="display: none;"
            >
              <g>
                <path
                  fill-opacity="0"
                  stroke-linecap="butt"
                  stroke-linejoin="miter"
                  stroke-miterlimit="4"
                />
              </g>
            </g>
            <g
              style="display: none;"
            >
              <g>
                <path
                  fill-opacity="0"
                  stroke-linecap="butt"
                  stroke-linejoin="miter"
                  stroke-miterlimit="4"
                />
              </g>
            </g>
            <g
              style="display: none;"
            >
              <g>
                <path
                  fill-opacity="0"
                  stroke-linecap="butt"
                  stroke-linejoin="miter"
                  stroke-miterlimit="4"
                />
              </g>
            </g>
          </g>
        </svg>
      </div>
    </div>
  </div>
</div>
`;
